src/commands/canBindGrammar.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ *  Copyright (c) 2021 Red Hat, Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-v20.html
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+
+'use strict';
+
+import { window, workspace, Disposable, TextDocumentContentChangeEvent, TextDocument, commands } from 'vscode';
+import { TextDocumentIdentifier } from 'vscode-languageclient';
+import { ServerCommandConstants } from './commandConstants';
+
+export function activateCanBindGrammar(supportedLanguages: { [id: string]: boolean }): Disposable {
+
+  let disposables: Disposable[] = [];
+  workspace.onDidChangeTextDocument(event => onDidChangeTextDocument(event.document, event.contentChanges), null, disposables);
+
+  let isEnabled = false;
+  updateEnabledState();
+  window.onDidChangeActiveTextEditor(updateEnabledState, null, disposables);
+
+  let timeout: NodeJS.Timer | undefined = undefined;
+
+  async function updateEnabledState() {
+    let editor = window.activeTextEditor;
+    if (!editor) {
+      return;
+    }
+    let document = editor.document;
+    if (!supportedLanguages[document.languageId]) {
+      return;
+    }
+
+    const documentURI= document.uri;
+    const identifier = TextDocumentIdentifier.create(documentURI.toString());
+    const result = await commands.executeCommand(ServerCommandConstants.CHECK_BOUND_GRAMMAR, identifier);
+    commands.executeCommand('setContext', 'canBindGrammar', result)
+  }
+
+  function onDidChangeTextDocument(document: TextDocument, changes: ReadonlyArray<TextDocumentContentChangeEvent>) {
```

Comment:
Remove this listener

Reply:
Without this listener, there's no way for the extension to know that the file is now bound is there?

Reply:
For example, if I manually remove the binding in the text editor